File: js/task1/S1_VAR_08.js
function safeAdd(a, b) {
  if (Number.isInteger(a) && Number.isInteger(b)) {
      if (Math.abs(a) > Number.MAX_SAFE_INTEGER || 
          Math.abs(b) > Number.MAX_SAFE_INTEGER) {

          const result = BigInt(a) + BigInt(b);
          console.log("bigint");
          return result;
      }

      const result = a + b;
      console.log("number");
      return result;
  }

  console.log("null");
  return null;
}

console.log(safeAdd(1, 2));                    
console.log(safeAdd(Number.MAX_SAFE_INTEGER, 1));  
console.log(safeAdd(-Number.MAX_SAFE_INTEGER, -1)); 
console.log(safeAdd(1.5, 2));                  
console.log(safeAdd("1", 2));                 
console.log(safeAdd(undefined, 2));

File: python/task1/S1_VAR_08.py
def safeAdd(a, b):
  if isinstance(a, int) and isinstance(b, int):
      if abs(a) > 2**53 - 1 or abs(b) > 2**53 - 1:
          print("bigint")
          return a + b
      print("number")
      return a + b

  print("null")
  return None

print(safeAdd(1, 2))
print(safeAdd(2**53 - 1, 1))
print(safeAdd(-(2**53 - 1), -1))
print(safeAdd(1.5, 2))
print(safeAdd("1", 2))
print(safeAdd(None, 2))
